server/src/controllers/deleteBooking.js

```diff
@@ -0,0 +1,30 @@
+const Boom = require('@hapi/boom');
+const { deleteBookingById, getBooking } = require('../database/queries');
+
+module.exports = (req, res, next) => {
+  const { id } = req.params;
+  const { userID, role } = req.user;
+
+  getBooking(id)
+    .then(({ rows }) => {
+      if (rows.length !== 0) {
+        return rows[0];
+      }
+      throw Boom.badRequest(
+        'the booking you are trying to delete does not exist'
+      );
+    })
+    .then((result) => {
+      if (role) {
+        return deleteBookingById(id);
+      }
+      const userId = result.user_id;
+      if (userId === userID) {
```

Comment:
variable names confusing, in these cases its fine to refer to them as `req.user.userID` and result.user_id